### PR TITLE
Clean up use of suppress_instrumentation in context and fix httpx bug

### PR DESCRIPTION
**Pull request by adriangb**
Currently, the exporters set the `_SUPPRESS_INSTRUMENTATION_KEY` key but httpx only checks for `"suppress_instrumentation"`.

This fixes that bug and does some cleanup to prevent similar bugs in the future.

### CODE REVIEW

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py [hunk 1]
```diff
@@ -152,3 +152,21 @@ def _python_path_without_directory(python_path, directory, path_separator):
...
+        or 
+        context.get_value(context._SUPPRESS_INSTRUMENTATION_KEY)  # type: ignore
+    ):
+        return False
+    return True
+
+
+def is_http_instrumentation_enabled() -> bool:
+    return (
+        is_http_instrumentation_enabled()
```
Comment (srikanthccv) [inline opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py:169]: did you mean `is_instrumentation_enabled`?

Comment (adriangb) [reply opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py:169]: Probably! Still need to run linting etc on this

File: instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py [hunk 2]
```diff
@@ -197,25 +195,22 @@ def _patched_api_call(self, original_func, instance, args, kwargs):
...
-            except ClientError as error:
-                result = getattr(error, "response", None)
-                _apply_response_attributes(span, result)
-                _safe_invoke(extension.on_error, span, error)
-                raise
-            else:
-                _apply_response_attributes(span, result)
-                _safe_invoke(extension.on_success, span, result)
+                with suppress_http_instrumentation():
+                    result = None
```
Comment (adriangb) [inline instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py:200]: I tried to preserve the logic with the exception catching and finally here but a second set of eyes would be good, especially given that I don't know if calling `_safe_invoke(extension.after_service_call)` after de-suppressing instrumentation is on purpose or it just happened to be like that

File: instrumentation/opentelemetry-instrumentation-cassandra/src/opentelemetry/instrumentation/cassandra/__init__.py [hunk 3]
```diff
@@ -65,7 +65,10 @@ def _traced_execute_async(func, instance, args, kwargs):
             if span.is_recording():
                 span.set_attribute(SpanAttributes.DB_NAME, instance.keyspace)
                 span.set_attribute(SpanAttributes.DB_SYSTEM, "cassandra")
-                span.set_attribute(SpanAttributes.NET_PEER_NAME, instance.cluster.contact_points)
+                span.set_attribute(
+                    SpanAttributes.NET_PEER_NAME,
+                    instance.cluster.contact_points,
+                )
```
Comment (adriangb) [inline instrumentation/opentelemetry-instrumentation-cassandra/src/opentelemetry/instrumentation/cassandra/__init__.py:71]: I ran `.tox/lint/bin/black .` after `tox -e lint` (which failed with `Could not find a version that satisfies the requirement opentelemetry-semantic-conventions==0.42b0.dev` for `opentelemetry-instrumentation-wsgi[test]`; I'm not sure how to point tox to my local version; I'm using `tox==3.27.1`) and it did some reformatting. I can try to revert this if I can get some pointers on the dependency issue.

Comment (adriangb) [reply instrumentation/opentelemetry-instrumentation-cassandra/src/opentelemetry/instrumentation/cassandra/__init__.py]: I manually reverted most of these for now

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py [hunk 4]
```diff
@@ -212,3 +217,44 @@ def _get_opentelemetry_stability_opt_in(
...
+        yield
+    finally:
+        context.detach(token)
+
+
+@contextmanager
+def suppress_instrumentation() -> Iterable[None]:
+    """Suppress instrumentation within the context."""
+    with _suppress_instrumentation(_SUPPRESS_INSTRUMENTATION_KEY):
+        yield
```
Comment (adriangb) [inline opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py:253]: We might want to also set the `"suppress_instrumentation"` (with no uuid) key here for any users that were relying on it.

**Comment by adriangb** [comment]
@srikanthccv would you mind reviewing again? Thanks!

**Comment by srikanthccv** [comment]
Thank you @adriangb. The changes are a little bigger than usual. I will take some time to review this one.

**Comment by adriangb** [comment]
No problem. I could make the diff smaller by e.g. adding the new functions and only replacing the old ones in one spot as a proof of concept then doing a followup PR that is just replacing other uses of the context.

**Comment by adriangb** [comment]
@srikanthccv let me know what you'd like to do with this PR so we don't get more conflicts to keep resolving

File: instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py (hunk 2 above)
Comment (srikanthccv) [reply instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py:203]: This looks fine

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py (hunk 4 above)
Comment (srikanthccv) [reply opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py:253]: We kept these suppress keys internal so far so it should be fine to not set that.

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**